code/game/machinery/doppler_array.dm

```diff
@@ -8,13 +8,29 @@ var/list/doppler_arrays = list()
 	density = 1
 	anchored = 1
 	atom_say_verb = "states coldly"
+	var/list/logged_explosions = list()
+
+/datum/explosion_log
+	var/logged_time
+	var/epicenter
+	var/actual_size_message
+	var/theoretical_size_message
+
+/datum/explosion_log/New(var/log_time, var/log_epicenter, var/log_actual_size_message, var/log_theoretical_size_message)
+	..()
+	logged_time = log_time
+	epicenter = log_epicenter
+	actual_size_message = log_actual_size_message
+	theoretical_size_message = log_theoretical_size_message
 
 /obj/machinery/doppler_array/New()
 	..()
 	doppler_arrays += src
 
 /obj/machinery/doppler_array/Destroy()
 	doppler_arrays -= src
+	for(var/datum/D in logged_explosions)
+		qdel(D)

```

Comment:
`logged_explosions.cut()`

Reply:
Do you want this to call `logged_explosions.cut()` before `qdel(D)`? or just `logged_explosions.cut()`?